src/matter/common/VendorId.ts

```diff
@@ -4,18 +4,18 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Typed, UnsignedIntT } from "../../codec/TlvObjectCodec";
+import { BoundedUnsignedIntT, Typed, UnsignedIntT } from "../../codec/TlvObjectCodec";
 import { MatterCoreSpecificationV1_0 } from "../../Specifications";
 
 /**
  * A Vendor Identifier (Vendor ID or VID) is a 16-bit number that uniquely identifies a particular
- * prod­uct manufacturer, vendor, or group thereof. Each Vendor ID is statically allocated by the
- * Connectiv­ity Standards Alliance (see [CSA Manufacturer Code Database]).
- * 
+ * product manufacturer, vendor, or group thereof. Each Vendor ID is statically allocated by the
+ * Connectivity Standards Alliance (see [CSA Manufacturer Code Database]).
+ *
  * @see {@link MatterCoreSpecificationV1_0} § 2.5.2
  */
 export type VendorId = { vendorId: true /* Hack to force strong type checking at compile time */ };
 export const VendorId = (id: number) => id as unknown as VendorId;
 
 /** Data model for a Vendor Identifier. */
-export const VendorIdT = Typed<VendorId>(UnsignedIntT);
+export const VendorIdT = Typed<VendorId>(BoundedUnsignedIntT({ min: 0, max: 0xFFFF }));
```

Comment:
I will add a PR to introduce UInt16T. Maybe you can merge and use it here

Reply:
Because PR not ready as of your info it is slightly delayed ... :-)

Reply:
@mfucci Do you have an answer here? (also for future work)